### PR TITLE
Improve integration of Custom Views in tabular pages

### PR DESCRIPTION
**Pull request by CarlosCortizasCT**
#### Summary

Improve integration of Custom Views in tabular pages

#### Description

The initial implementation for integrating the upcoming Custom Views feature in tabular pages components only allowed to provide a single locator for the page, where this kind are supposed to have different exclusive sections (which also have different URLs) so it does not support specifying a locator for an specific tab and this is something needed.

In this PR we refactor the integration so tab page components consumers can provide a map of the locators based on the actual URL (the same mapping it's used for deciding which tab content to render).

```jsx
// BEFORE
<TabularMainPage
  title="Product"
  customViewLocatorCode="products.product_variant_details.general">

// AFTER
<TabularMainPage
  title="Product"
  customViewLocatorCodes={{
    'products.product_variant_details.general': `${match.url}`,
    'products.product_variant_details.variants': `${match.url}/variants`,
    'products.product_variant_details.search': `${match.url}/search`
  }}
>
```

### CODE REVIEW

**Review by emmenko — COMMENTED** [review]
Ah good catch! I like the idea! 💯

File: packages/application-components/src/hooks/use-custom-view-locator-selector/use-custom-view-locator-selector.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        return matchPath(location.pathname, {
+          // strip the search, otherwise the path won't match
+          path: pathWithoutSearch(locator),
+          exact: false,
+          strict: false,
+        });
+      }
+    );
+    setCurrentCustomViewLocatorCode(customViewLocator?.[0]);
+  }, [customViewLocatorCodes, location]);
```
[the remark this answers is not in the document]
Comment (emmenko) [reply packages/application-components/src/hooks/use-custom-view-locator-selector/use-custom-view-locator-selector.ts:27]: Using `useEffect`, `useState` etc is always a bit tricky as you need to be careful of potential re-renders. In particular with `useEffect` there is always a risk of ending up in an infinite loop.

If it's not absolutely necessary, I would keep it simple.

As for the "optimizations", matching a path or multiple ones is really a "cheap" operation (you can see the implementation of [matchPath](https://github.com/remix-run/react-router/blob/v5.3.4/packages/react-router/modules/matchPath.js), it's just simple JS).
However, if you really do want to optimize the calculation I would suggest to use `useMemo` as it's more suited for this use case.

[the remark this answers is not in the document]
Comment (CarlosCortizasCT) [reply packages/application-components/src/hooks/use-custom-view-locator-selector/use-custom-view-locator-selector.ts]: I don't see the potential of an infinite loop here and I'm used to avoiding lists iterations whenever I can, but I agree here it's a non-worthy optimization as the list won't ever be too large.

Refactored here: c94cb17befedf5dd3c98b566dde42fb3ffd637b9

**Review by emmenko — APPROVED** [review]
💯 Thanks

**Pull request merged**